src/Common/Standards/OgcApi.Net.Styles/Storage/FileSystem/StyleFileSystemStorage.cs

```diff
@@ -0,0 +1,197 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Options;
+using OgcApi.Net.Resources;
+using OgcApi.Net.Styles.Model.Metadata;
+using OgcApi.Net.Styles.Model.Styles;
+using OgcApi.Net.Styles.Model.Stylesheets;
+using System.Collections.Concurrent;
+using System.IO;
+using System.Text.Json;
+
+namespace OgcApi.Net.Styles.Storage.FileSystem;
+
+public class StyleFileSystemStorage(IOptionsMonitor<StyleFileSystemStorageOptions> options) : IStyleStorage
+{
+    private readonly StyleFileSystemStorageOptions _options = options.CurrentValue;
+    private static readonly ConcurrentDictionary<string, object> _locks = new();
```

Comment:
This dictionary is never cleared, which can lead to high memory consumption. To do this, it is enough to send requests to the API to add a style with random identifiers.